maestro-cli/src/main/java/maestro/cli/device/DeviceCreateUtil.kt

```diff
@@ -0,0 +1,145 @@
+package maestro.cli.device
+
+import maestro.cli.CliError
+import maestro.cli.util.DeviceConfigAndroid
+import maestro.cli.util.DeviceConfigIos
+import maestro.cli.util.PrintUtils
+
+internal object DeviceCreateUtil {
+
+    fun getOrCreateDevice(platform: Platform, osVersion: Int?, forceCreate: Boolean): Device.AvailableForLaunch {
+        return when (platform) {
+            Platform.ANDROID -> {
+                getOrCreateAndroidDevice(osVersion, forceCreate)
+            }
+
+            Platform.IOS -> {
+                getOrCreateIosDevice(osVersion, forceCreate)
+            }
+
+            else -> throw CliError("Unsupported platform $platform. Please one of: android, ios")
+        }
+    }
+
+    private fun getOrCreateIosDevice(version: Int?, forceCreate: Boolean): Device.AvailableForLaunch {
+        if (version !in DeviceConfigIos.versions) {
+            throw CliError("Provided iOS version is not supported. Please use one of ${DeviceConfigIos.versions}")
+        }
+
+        val runtime = DeviceConfigIos.runtimes[version]
+        if (runtime == null) {
+            throw CliError("Provided iOS runtime is not supported $runtime")
+        }
+
+        val deviceName = DeviceConfigIos.generateDeviceName(version!!)
+        val device = DeviceConfigIos.device
+
+        // check connected device
+        if (DeviceService.isDeviceConnected(deviceName, Platform.IOS) != null) {
+            throw CliError("A device with name $deviceName is already connected")
+        }
+
+        // check existing device
+        val existingDeviceId = DeviceService.isDeviceAvailableToLaunch(deviceName, Platform.IOS)?.let {
+            if (forceCreate) {
+                DeviceService.deleteIosDevice(it.modelId)
+                null
+            } else it.modelId
+        }
+
+        if (existingDeviceId != null) PrintUtils.message("Using existing device $deviceName (${existingDeviceId}).")
+        else PrintUtils.message("Attempting to create iOS simulator: $deviceName ")
+
+
+        val deviceUUID = try {
+            existingDeviceId ?: DeviceService.createIosDevice(deviceName, device, runtime).toString()
+        } catch (e: IllegalStateException) {
+            val error = e.message ?: ""
+            if (error.contains("Invalid runtime")) {
+                val msg = "Required runtime to create the simulator is not installed: $runtime\n\n" +
+                        "To install additional iOS runtimes checkout this guide:\n" +
+                        "* https://developer.apple.com/documentation/xcode/installing-additional-simulator-runtimes"
+                throw CliError(msg)
+            } else if (error.contains("Invalid device type")) {
+                throw CliError("Device type $device is either not supported or not found.")
```

Comment:
Is there a way we can guide users to recover from this error?

Reply:
Ideally, this should never happen. From what I know, Xcode has all the devices available without having to install additional components.

This is to cover edge cases.